HGustavs/LenaSYS
# G2 2023 v4 issue #13532 Created test API

Related to #13532 

Created test.php file in /Shared folder which helps testing services.
This includes 3 tests.
* Assert test (Compare two values)
* Login test - Check if login credentials are valid
* Call service - With the help of parameters like opt some service functionality can be executed

Session.php have been modified to allow checkLogin() with POST

----------------------------------------------------------
## Example on how to run this test from another file:

include_once "../Shared/test.php";

$testData = array(
    'value1' => 'test',
    'value2' => 'test',
    'service' => 'https://cms.webug.se/root/G2/students/c21alest/LenaSYS/DuggaSys/courseedservice.php',
    'serviceData' => serialize(array( // Data that service needs to execute function
        'opt' => 'NEW',
        'username' => 'usr',
        'password' => 'pass',
    )),
    'test1Debug' => false, // If true more information of test will be displayed
    'test2Debug' => true,
    'test3Debug' => true,
);

testHandler($testData, false); // 2nd argument true = prettyprint (HTML) false = raw JSON, no debug mode is avalible when using prettyprint

----------------------------------------------------------

### To test
* I have not yet checked that a service call makes any changes to the database, just that the service marks it as a success.
* Check that different login types are working (student, teacher etc) and that a student can't perform any activities that requires a teacher or admin.
* Test that JSON format can be compared in the assert test.
* Is it clear how to use the test or is any further documentation needed?
* Session.php has been modified so further testing might be needed to make sure that this won't brake anything
